Makefile

```diff
@@ -271,10 +271,11 @@ unset-override:
 
 pre-format: unset-override
 	@rustup component add rustfmt
+	@cargo install -q cargo-sort 
 
 format: pre-format
-	@cargo fmt -- --check >/dev/null || \
-	cargo fmt
+	@cargo fmt -- --check >/dev/null || cargo fmt
```

Comment:
The first check seems useless.

Reply:
The first check is to do the actual format, and the second is to print out the error if there is something wrong with the `cargo fmt`

Reply:
No, the check won't do format. It prints unformatted code to stdout and leave it at that.

I think this line first scans the code in read-only mode, and then do the actual format if needed. But it's not like "format" has a big overhead. I don't see why we need to check in read-only mode first.

Reply:
I see